EE.C

```diff
@@ -0,0 +1,478 @@
+//      root> t.Loop();       // Loop on all entries
+//
+
+//     This is the loop skeleton where:
+//    jentry is the global entry number in the chain
+//    ientry is the entry number in the current Tree
+//  Note that the argument to GetEntry must be:
+//    jentry for TChain::GetEntry
+//    ientry for DYree::GetEntry and TBranch::GetEntry
+//
+//       To read only selected branches, Insert statements like:
+// METHOD1:
+//    fChain->SetBranchStatus("*",0);  // disable all branches
+//    fChain->SetBranchStatus("branchname",1);  // activate branchname
+// METHOD2: replace line
+//    fChain->GetEntry(jentry);       //read all branches
+//by  b_branchname->GetEntry(ientry); //read only this branch
+   if (fChain == 0) return;
+
+   TFile *output = new TFile("DY_EE.root","recreate");
+
+   TH1F *run_DY = new TH1F("run_DY","run",3,0,3);
+   TH1F *leppT_DY = new TH1F("leppT_DY","lepton pt",100,0,500);
+   
+   TH1F *lepeta_DY = new TH1F("lepeta_DY","lepton eta",20,-3,3);
+   
+   TH1F *lepphi_DY = new TH1F("lepphi_DY","lepton phi",20,-3.5,3.5);
+   
+   TH1F *dPhi_DY = new TH1F("dPhi_DY","delta phi_{e,#mu}",50,-4,4);
+   TH1F *dR_DY = new TH1F("dR_DY","delta R_{e,#mu}",50,0,5);
+   TH1F *ee_mass_DY = new TH1F("ee_mass_DY","Mass_{e,e}",50,0,500);
+  
+   TH1F *eleIso_DY = new TH1F("eleIso_DY","electron isolation",100,0,400);
+
+   TH1F *jetpT_DY = new TH1F("jetpT_DY","jet pT",100,0,500);
+   TH1F *jeteta_DY = new TH1F("jeteta_DY","jet eta",20,-5,5);
+   
+   TH1F *bjetpT_DY = new TH1F("bjetpT_DY","b-jet pT",100,0,500);
+   TH1F *bjeteta_DY = new TH1F("bjeteta_DY","b-jet eta",20,-5,5);
+
+   TH1F *LeadJetpT_DY = new TH1F("LeadJetpT_DY","Leading b-jet pT",50,0,500);
+   TH1F *SecJetpT_DY = new TH1F("SecJetpT_DY","2nd leading b-jet pT",50,0,500);
+   
+//   TH1F *btagEvent_DY = new TH1F("btagEvent_DY","btagEvent",100,0,500);
+
+   TH1F *njet_DY = new TH1F("njet_DY","Number of jet",11,0,11);
+   TH1F *nbjet_DY = new TH1F("nbjet_DY","Number of b-jet",11,0,11);
+   TH1F *id_DY = new TH1F("id_DY","particle id",30,-15,15);
+   
+   TH1F *MET_DY = new TH1F("MET_DY","Missing ET",100,0,1000);
+   TH1F *dilep_mass_DY = new TH1F("dilep_mass_DY","dilepton mass",100,0,1000);
+   
+   TH1F *SBM_DY = new TH1F("SBM_DY","max(SBM)",100,0,300);
+   TH1F *METvsMmm_DY = new TH1F("METvsMmm_DY","E^{miss}_{T}/M(#mu^{+}#mu^{-})",100,0,1);
+   TH1F *HTLT_DY = new TH1F("HTLT_DY","HT-LT",100,-500,500);
+
+   TH1F *Mass_DY = new TH1F("Mass_DY","Lepton mass(M_{e})",100,0,500);
+   
+   TH2F *njvsnbj_DY = new TH2F("njvsnbj_DY","njet vs nbjet",10,0,10,5,0,5);
+//   TH2F *bjetpT_dR_DY = new TH2F("bjetpT_dR_DY","b-jet pT vs DeltaR",100,0,500,50,0,5);
+//   TH2F *bjetpT_elepT_DY = new TH2F("bjetpT_elepT_DY","b-jet pT vs electron pT",100,0,500,100,0,500);
+//   TH2F *bjetpT_mupT_DY = new TH2F("bjetpT_mupT_DY","b-jet pT vs muon pT",100,0,500,100,0,500);
+   
+   Long64_t nentries = fChain->GetEntriesFast();
+
+   float totalN = 122055388;
```

Comment:
Do you change this by hand for every sample? It would be better to declare these values as general quantities in the .h of this class, then set them as a response to inputting a filename or descriptor for the file to be loaded alongside the relevant dataset.
That avoids nasty errors due to changing numbers by hand all the time.